vars/pod.groovy

```diff
@@ -80,11 +80,18 @@ def call(params = [:], Closure body) {
     }
 
     params['secrets'].eachWithIndex { secret, i ->
-        podObj['spec']['volumes'] += ['name': "secret-${i}".toString(), 'secret': [secretName: secret]]
-        podObj['spec']['containers'][0]['volumeMounts'] += ['name': "secret-${i}".toString(), 'mountPath': "/run/kubernetes/secrets/${secret}".toString()]
-
+        if (secret.contains(":")) {
+            def key
+            (secret, key, mountPath) = secret.split(":")
+            podObj['spec']['volumes'] += ['name': secret, 'secret': [secretName: secret, items: [['key':key, 'path': key]]]]
```

Comment:
```suggestion
            podObj['spec']['volumes'] += ['name': secret, 'secret': [secretName: secret, items: [['key': key, 'path': key]]]]
```

Reply:
Multiple keys from the same secret may be used multiple times. So for the secret name, maybe it's best to keep it `secret-${i}`?